# King County Housing Data Analysis

#### First let's import the tools we are going to need to analyse our dataset

In [78]:
import pandas as pd
import seaborn as sns

%matplotlib inline

#### Let's load our dataset and preview it!

In [79]:
df = pd.read_csv("../kc_house_data.csv")
df.head()

id        date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  10/13/2014  221900.0         3       1.00         1180   
1  6414100192   12/9/2014  538000.0         3       2.25         2570   
2  5631500400   2/25/2015  180000.0         2       1.00          770   
3  2487200875   12/9/2014  604000.0         4       3.00         1960   
4  1954400510   2/18/2015  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0         NaN   0.0  ...      7        1180            0.0   
1      7242     2.0         0.0   0.0  ...      7        2170          400.0   
2     10000     1.0         0.0   0.0  ...      6         770            0.0   
3      5000     1.0         0.0   0.0  ...      7        1050          910.0   
4      8080     1.0         0.0   0.0  ...      8        1680            0.0   

  yr_built  yr_renovated  zipcode      lat     long  sqft_living15  sqft_lot15  
0     1955           0.0    98178  47.5112 -122.257           1340        5650  
1     1951        1991.0    98125  47.7210 -122.319           1690        7639  
2     1933           NaN    98028  47.7379 -122.233           2720        8062  
3     1965           0.0    98136  47.5208 -122.393           1360        5000  
4     1987           0.0    98074  47.6168 -122.045           1800        7503  

[5 rows x 21 columns]

#### Now we'll get the main characteristics of our dataset

In [80]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21597 entries, 0 to 21596
Data columns (total 21 columns):
id               21597 non-null int64
date             21597 non-null object
price            21597 non-null float64
bedrooms         21597 non-null int64
bathrooms        21597 non-null float64
sqft_living      21597 non-null int64
sqft_lot         21597 non-null int64
floors           21597 non-null float64
waterfront       19221 non-null float64
view             21534 non-null float64
condition        21597 non-null int64
grade            21597 non-null int64
sqft_above       21597 non-null int64
sqft_basement    21597 non-null object
yr_built         21597 non-null int64
yr_renovated     17755 non-null float64
zipcode          21597 non-null int64
lat              21597 non-null float64
long             21597 non-null float64
sqft_living15    21597 non-null int64
sqft_lot15       21597 non-null int64
dtypes: float64(8), int64(11), object(2)
memory usage: 3.5+ MB
None


## Now it's time to clean our dataset

We observed that 3 of our columns are missing values, namely "waterfront", "view" and "yr_renovated". Let's have a more precise look at each one.

##### Cleaning the waterfront column:

In [81]:
print("We have " + str(df["waterfront"].isna().sum()) + " missing values for the waterfront column.\n")
print("Here are the value counts for this column:")
print(df["waterfront"].value_counts())

We have 2376 missing values for the waterfront column.

Here are the value counts for this column:
0.0    19075
1.0      146
Name: waterfront, dtype: int64


We observe that the waterfront feature is basically a boolean value and only a very small percentage
of the houses have this feature, so it make sens to replace the rows missing waterfront's value by 0.

In [82]:
df.loc[df["waterfront"].isna(), "waterfront"] = 0
print("Here are the updated value counts for the waterfront column:")
print(df["waterfront"].value_counts())

Here are the updated value counts for the waterfront column:
0.0    21451
1.0      146
Name: waterfront, dtype: int64


##### Cleaning the view column:

In [83]:
print("We have " + str(df["view"].isna().sum()) + " missing values for the view column.\n")
print(df["view"].value_counts())


We have 63 missing values for the view column.

0.0    19422
2.0      957
3.0      508
1.0      330
4.0      317
Name: view, dtype: int64


We observe that the very large majority of the view feature is 0 and since we have only 63 missing values it make sens to replace the missing values by 0 (the mode).
We chose to use the mode because even though the column format is a float, the numbers could be seen as categorical.
Doing this will barely change our average or median for this column.

In [84]:
df.loc[df["view"].isna(), "view"] = 0
print("Here are the updated value counts for the view column:")
print(df["view"].value_counts())

Here are the updated value counts for the view column:
0.0    19485
2.0      957
3.0      508
1.0      330
4.0      317
Name: view, dtype: int64


##### Cleaning the yr_renovated column:

In [85]:
print("We have " + str(df["yr_renovated"].isna().sum()) + " missing values for the yr_renovated column.\n")
df["yr_renovated"].value_counts().head()
# In total 744 house from our dataset have been renovated
# It make sens to keep this data and turn it into a boolean category
### Most houses seems to have been renovated in the same decade so keeping the date doesnt add any value ###
### Should we consider house renovated more than 25 years ago renovated ? ###

We have 3842 missing values for the yr_renovated column.



0.0       17011
2014.0       73
2003.0       31
2013.0       31
2007.0       30
Name: yr_renovated, dtype: int64

We observe that 3842 + 17011 values are either missing or have the value of 0.
We might think about dropping the whole column but it could later eveal to be a useful information, instead we should treat as categorical data.
### Most houses seems to have been renovated in the same decade so keeping the date doesnt add any value ###
### Should we consider house renovated more than 25 years ago renovated ? ###

In [86]:
df.loc[(df["yr_renovated"].isna() == True), "yr_renovated"] = False
df.loc[(df["yr_renovated"] == 0.0), "yr_renovated"] = False
# We are considering that houses renovated before 1990 should not be considered as renovated
df.loc[df["yr_renovated"] < 1990, "yr_renovated"] = False
# We now changing the year into True
df.loc[df["yr_renovated"] > 0, "yr_renovated"] = True
print(df["yr_renovated"].value_counts())

False    21075
True       522
Name: yr_renovated, dtype: int64
